### PR TITLE
Api

### PR DESCRIPTION
**Pull request by obed300**
In this project, I connected my React & Redux application to an existing API to create and remove books from a server

### CODE REVIEW

File: src/components/Displaybooks.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Book from './Book';
+import { fetchBooks } from '../features/book/Bookslice';
+
+function Displaybooks() {
+  const { books } = useSelector((state) => state.book);
+  //   const { categories } = useSelector((state) => state.categories);
+  const dispatch = useDispatch();
```
Comment (AmaduKamara) [inline src/components/Displaybooks.js]: - [ ] Let's try not to have commented unused code in our codebase as they make our code dirty and not professional. Kindly remove all unused codes from your codebase 👍🏽

File: src/components/Form.js [hunk 2]
```diff
@@ -1,14 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+// import { nanoid } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
```
Comment (AmaduKamara) [inline src/components/Form.js]: - [ ] Let's try not to have commented unused code in our codebase as they make our code dirty and not professional. Kindly remove all unused codes from your codebase 👍🏽

File: src/components/Form.js [hunk 3]
```diff
@@ -1,14 +1,55 @@
...
+      item_id: uuidv4(),
+      title,
+      author,
+
+    };
+    dispatch(addBook(bookData));
+    dispatch(bookPost(bookData));
+    setTitle('');
+    setAuthor('');
+  };
```
Comment (AmaduKamara) [inline src/components/Form.js]: - [ ] When I add a book, it gets added to the store and rendered on the screen, but when the browser is reloaded with the expectation of the data being persisted to the API, the data is not saved to the API. Kindly check that out and be sure any book added is saved to the API, to the store, and is rendered as expected 👍🏽

File: src/features/book/Bookslice.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
```
Comment (AmaduKamara) [inline src/features/book/Bookslice.js]: - [ ] Let's try to not disable Linter checks on our codes as they help us write better and more professional clean code. Disabling Linter checks defies its purpose. Please fix the error on your next submission. 👍🏽

File: src/features/book/Bookslice.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  },
+);
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      // const bookData = actions.payload;
+      state.books.push(action.payload);
```
Comment (AmaduKamara) [inline src/features/book/Bookslice.js]: - [ ] Let's try not to have commented unused code in our codebase as they make our code dirty and not professional. Kindly remove all unused codes from your codebase 👍🏽

File: src/features/book/Bookslice.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+    builder.addCase(fetchBooks.pending, (state) => {
+      state.isLoading = true;
+    }).addCase(fetchBooks.fulfilled, (state, action) => {
+      state.isLoading = false;
+      const responseObj = action.payload;
+
+      const newBookArr = [];
+      // eslint-disable-next-line no-restricted-syntax, guard-for-in
+      for (const id in responseObj) {
```
Comment (AmaduKamara) [inline src/features/book/Bookslice.js]: - [ ] Let's try to not disable Linter checks on our codes as they help us write better and more professional clean code. Disabling Linter checks defies its purpose. Please fix the error on your next submission. 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @obed300,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽 
✅ Linter checks passed 👍🏽
✅ Used Redux Toolkit and createAsyncThunk API as requested 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly modify your README file to have all contents related to the bookstore app. Currently, you are having some content related to the Math Magician app. Please modify this file to have the correct documentation for this app 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Form.js (hunk 3 above)
Comment (rloterh) [reply src/components/Form.js]: resolved! ✔️

File: src/features/book/Bookslice.js (hunk 4 above)
Comment (rloterh) [reply src/features/book/Bookslice.js]: resolved! ✔️

File: src/features/book/Bookslice.js (hunk 5 above)
Comment (rloterh) [reply src/features/book/Bookslice.js]: resolved! ✔️

File: src/features/book/Bookslice.js (hunk 6 above)
Comment (rloterh) [reply src/features/book/Bookslice.js]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Obed,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Good job implementing recommended changes!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Kindly ensure that the _Getting Started_ section_ is filled with the appropriate instructions on how to _setup_ and _run_ your app. Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**